Mathlib/NumberTheory/LSeries/HurwitzZetaEven.lean

```diff
@@ -0,0 +1,386 @@
+      ofReal_one, ofReal_ofNat]
+  have h4 : -π * I * (a * I * x) ^ 2 / (I * x) = - (-π * a ^ 2 * x) := by
+    rw [mul_pow, mul_pow, I_sq]; field_simp [hx.ne']; ring_nf
+  rw [h1, h2, h3, h4, ← mul_assoc, mul_comm (cexp _), mul_assoc _ (cexp _) (cexp _),
+    ← Complex.exp_add, neg_add_self, Complex.exp_zero, mul_one, ofReal_div, ofReal_one]
+
+end kernel_defs
+
+section asymp
+
+/-!
+## Formulae for the kernels as sums
+-/
+
+lemma hasSum_int_evenKernel (a : ℝ) {t : ℝ} (ht : 0 < t) :
+    HasSum (fun n : ℤ ↦ rexp (-π * (n + a) ^ 2 * t)) (evenKernel a t) := by
+  rw [← hasSum_ofReal, evenKernel_def]
+  convert (hasSum_jacobiTheta₂_term _ (by rwa [I_mul_im, ofReal_re])).mul_left _
+    using 2 with n
+  rw [jacobiTheta₂_term, ← Complex.exp_add]
+  push_cast
+  ring_nf
+  simp only [I_sq, mul_neg, neg_mul, mul_one]
+
+lemma hasSum_int_cosKernel (a : ℝ) {t : ℝ} (ht : 0 < t) :
+    HasSum (fun n : ℤ ↦ cexp (2 * π * I * a * n) * rexp (-π * n ^ 2 * t)) ↑(cosKernel a t) := by
+  rw [cosKernel_def a t]
+  convert hasSum_jacobiTheta₂_term a (by rwa [I_mul_im, ofReal_re] : 0 < im (I * t)) using 2 with n
+  rw [jacobiTheta₂_term, ofReal_exp, ← Complex.exp_add]
+  push_cast
+  ring_nf
+  simp only [I_sq, mul_neg, neg_mul, mul_one, sub_eq_add_neg]
+
+/-- Modified version of `hasSum_int_evenKernel` omitting the constant term at `∞`. -/
+lemma hasSum_int_evenKernel₀ (a : ℝ) {t : ℝ} (ht : 0 < t) :
+    HasSum (fun n : ℤ ↦ if n + a = 0 then 0 else rexp (-π * (n + a) ^ 2 * t))
+    (evenKernel a t - if (a : UnitAddCircle) = 0 then 1 else 0) := by
+  simp_rw [AddCircle.coe_eq_zero_iff, zsmul_one]
+  split_ifs with h
+  · obtain ⟨k, rfl⟩ := h
+    simp_rw [← Int.cast_add, Int.cast_eq_zero, add_eq_zero_iff_eq_neg]
+    simpa using hasSum_ite_sub_hasSum (hasSum_int_evenKernel (k : ℝ) ht) (-k)
+  · suffices ∀ (n : ℤ), n + a ≠ 0 by simpa [this] using hasSum_int_evenKernel a ht
+    contrapose! h
+    let ⟨n, hn⟩ := h
+    exact ⟨-n, by rwa [Int.cast_neg, neg_eq_iff_add_eq_zero]⟩
+
+lemma hasSum_int_cosKernel₀ (a : ℝ) {t : ℝ} (ht : 0 < t) :
+    HasSum (fun n : ℤ ↦ if n = 0 then 0 else cexp (2 * π * I * a * n) * rexp (-π * n ^ 2 * t))
+    (↑(cosKernel a t) - 1) := by simpa using hasSum_ite_sub_hasSum (hasSum_int_cosKernel a ht) 0
+
+lemma hasSum_nat_cosKernel₀ (a : ℝ) {t : ℝ} (ht : 0 < t) :
+    HasSum (fun n : ℕ ↦ 2 * Real.cos (2 * π * a * (n + 1)) * rexp (-π * (n + 1) ^ 2 * t))
+    (cosKernel a t - 1) := by
+  rw [← hasSum_ofReal, ofReal_sub, ofReal_one]
+  have := (hasSum_int_cosKernel a ht).sum_nat_of_sum_int
+  rw [← hasSum_nat_add_iff' 1] at this
+  simp_rw [Finset.sum_range_one, Nat.cast_zero, neg_zero, Int.cast_zero, zero_pow two_ne_zero,
+    mul_zero, zero_mul, Complex.exp_zero, Real.exp_zero, ofReal_one, mul_one, Int.cast_neg,
+    Int.cast_natCast, neg_sq, ← add_mul, add_sub_assoc, ← sub_sub, sub_self, zero_sub,
+    ← sub_eq_add_neg, mul_neg] at this
+  convert this with n
```

Comment:
This ` convert`  is also slow, but the resulting equation is maybe too long to state explicitly...